pkg/registry/core/service/rest_test.go

```diff
@@ -672,6 +627,19 @@ func TestServiceRegistryResourceLocation(t *testing.T) {
 	}
 }
 
+func checkLocation(r rest.Redirector, ctx api.Context, id, e string, t *testing.T) {
+	location, _, err := r.ResourceLocation(ctx, id)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
```

Comment:
If you make this change, you need to change the error messages so that when this fails, we will be able to tell which invocation of it failed.